frontend/src/AppRoot.tsx

```diff
@@ -22,82 +25,69 @@ import ROUTES from './utils/routes'
 
 const isDev = process.env.NODE_ENV === 'development'
 
-const fetchProfile = (token: string) => {
-  return fetch('/profile', {
-    headers: { Authorization: `Bearer ${token}` },
-  })
-}
-
 const AppRoot = () => {
-  const { isAuthenticated, isLoading, getAccessTokenSilently } = useAuth0()
-  const [profileIsLoaded, setProfileIsLoaded] = useState(false)
-
-  useEffect(() => {
-    if (isAuthenticated && !profileIsLoaded) {
-      getAccessTokenSilently()
-        .then(fetchProfile)
-        .then((response) => {
-          if (response.ok) {
-            setProfileIsLoaded(true)
-          } else {
-            console.error('Non-OK server response retrieving profile')
-          }
-        })
-    }
-  }, [isAuthenticated, getAccessTokenSilently, profileIsLoaded])
+  const { isLoading, isAuthenticated } = useAuth0()
 
   return (
     <ApolloAuthProvider>
-      <Router>
-        <Switch>
-          {isDev && (
-            <Route path={ROUTES.KITCHEN_SINK}>
-              <KitchenSink />
+      <UserProfileProvider>
```

Comment:
I wrapped the `/profile` call into its own Context component. This way, it's isolated into its own file, and we can access the user's profile from anywhere in the app.

Reply:
ah that's a great improvement